ls1intum/Artemis
# `Plagiarism checks`: Integrate course wide plagiarism handling with notifications

This PR allows instructors to handle accepted plagiarism cases in a new course-wide overview. Instructors can provide explanations about the case and notify students. Students can then see the plagiarism case in Artemis. They can then react to the accusation with an explanation in case they think the accusation is wrong.




### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This allows instructors to notify students to about potential plagiarism cases. The students can give statements to the plagiarism case and the instructor can decide whether to deny or confirm them.

### Description


### Steps for Testing

Prerequisites:

1 Instructor
2 Students
1 Text oder modeling exercise with a detected plagiarism
1a.Log in to Artemis as Instructor
1b. Navigate to Course Detail
1c. Open Plagiarism Cases
1d. Asses the plagiarism case and save instructor statement

2a. Log in to Artemis as Student
2b. Open Notifications
2c. Click on the Plagiarism Notification
2d. Respond to the plagiarism Case

3a. Login in again to Artemis as Instructor
3b. Navigate to Course Detail
3c. Open Plagiarism Cases
3d. Assess the student's response

4a. Log in to Artemis again as Student
4b. Open Notifications
4c. Click on the Plagiarism Notification
4d. Check that the status has been updated by the instructor

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





PlagiarismResource.java: 60% | 87% 
PlagiarismService.java: 37% | 70%
plagiarism-cases.service.ts: 67% | 100%
plagiarism-cases.component.ts: 50% | 77%
plagiarism-cases-review.component.ts: 87% | 95%
plagiarism-cases-list.component.ts: 93% | 100%

(Server test coverage will be improved in future as existing functionality wasn't tested before)

### Screenshots


Student viewing plagiarism case:
<img width="1431" alt="Bildschirmfoto 2021-11-14 um 12 39 39" src="https://user-images.githubusercontent.com/51987021/141679274-5663983d-7153-4971-9e8b-4031193a1e8d.png">
Instructor's view:
<img width="1431" alt="Bildschirmfoto 2021-11-14 um 12 39 08" src="https://user-images.githubusercontent.com/51987021/141679276-1ab85fe0-8d02-419c-bbd5-253d9f32edd8.png">
Button for instructor in course detail view to view all cases:
<img width="261" alt="Bildschirmfoto 2021-11-14 um 12 38 52" src="https://user-images.githubusercontent.com/51987021/141679277-e0a35852-b7dd-49dc-ba36-3d555506982a.png">
Instructor view of all cases:
![Bildschirmfoto 2021-12-11 um 13 34 36](https://user-images.githubusercontent.com/51987021/145678255-35de72ce-dba8-4901-b28c-ec76a0793370.png)
Notification for final decision:
<img width="301" alt="Bildschirmfoto 2021-11-14 um 12 38 38" src="https://user-images.githubusercontent.com/51987021/141679278-1326b08a-d34e-4f47-bdcd-09c3682ccb5f.png">
Notification about possible case:
![Bildschirmfoto 2021-12-11 um 13 33 27](https://user-images.githubusercontent.com/51987021/145676724-14fc0af7-1315-47e0-b1ea-ba69c88db92b.png)
Student's statement:
![Bildschirmfoto 2021-12-11 um 13 33 47](https://user-images.githubusercontent.com/51987021/145676703-89a477d8-d21e-4a4d-8e5c-04593a2fea6b.png)
